### PR TITLE
Add wildcard check to bans check

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

**Comment by Jake-Shadle** [comment]
I will try and look at this tomorrow, sorry for the wait!

**Comment by khodzha** [comment]
no problem, take your time! :smiley:

File: src/bans/mod.rs [hunk 1]
```diff
@@ -194,6 +194,8 @@ pub fn check(
     output_graph: Option<Box<OutputGraph>>,
     sender: crossbeam::channel::Sender<Pack>,
 ) {
+    let wildcard = VersionReq::parse("*").expect("Parsing wildcard mustnt fail");
```
Comment (Jake-Shadle) [inline src/bans/mod.rs:197]:
```suggestion
    let wildcard = VersionReq::any();
```

**Comment by khodzha** [comment]
rebased on current main + feedback addressed

File: src/bans/mod.rs (hunk 1 above)
Comment (khodzha) [reply src/bans/mod.rs:197]: hmm, it turns out `VersionReq::any()` and `VersionReq` with `Wildcard(major)` predicate fail equality comparison

Comment (Jake-Shadle) [reply src/bans/mod.rs:197]: Hmm ok...nvm then!

**Comment by repi** [comment]
This seems to generate wildcard warnings when using standard `path` dependencies and specifying no version (as one is using the local file version), don't think that should generate warnings.

Such as:

```toml
[dependencies]
telemetry = { path = "../telemetry" }
```

@Jake-Shadle

**Pull request merged**